lightningd/peer_control.c

```diff
@@ -871,30 +870,30 @@ static enum watch_result funding_depth_cb(struct lightningd *ld,
 			return DELETE_WATCH;
 		}
 
+		/* If we restart, we could already have peer->scid from database */
+		if (!channel->scid)
+			channel->scid = tal(channel, struct short_channel_id);
+		else if (!short_channel_id_eq(channel->scid, &scid)) {
+			/* Will restart channeld, initialized with updated scid, which
+			 * should add it (at least our halve of the channel) to rtable. */
+			channel_fail_transient(channel, "short_channel_id changed to %s (was %s)",
+					        short_channel_id_to_str(tmpctx, &scid),
+					        short_channel_id_to_str(tmpctx, channel->scid));
+		}
+
 		/* We've added scid, update */
+		*channel->scid = scid;
```

Comment:
I guess you want to ask Channeld to die after setting `*channel->scid = scid;` and saving into DB?
what about return immediately in` else if` block?
```
if (!channel->scid) {
	channel->scid = tal(channel, struct short_channel_id);
	*channel->scid = scid;
	wallet_channel_save(ld->wallet, channel);
} else if (!short_channel_id_eq(channel->scid, &scid)) {
	/* Will restart channeld, initialized with updated scid, which
	 * should add it (at least our halve of the channel) to rtable. */
	log_info(channel->log, "short_channel_id change %s -> %s",
			        short_channel_id_to_str(tmpctx, &scid),
			        short_channel_id_to_str(tmpctx, channel->scid));
	*channel->scid = scid;
	channel_fail_transient(channel, "Different scid, and reorg may happen. Restart channeld.");
	wallet_channel_save(ld->wallet, channel);
	return KEEP_WATCHING;
}
```

Reply:
Yes maybe that is more straight-forward (instead of having to go through `channel_tell_depth` first). Will fix that.

Reply:
With respect to `log_info`, I try to keep it concise and a `channel_fail_transient` is already logged at `log-level=info`.